codegen/sdk-codegen/build.gradle.kts

```diff
@@ -221,7 +218,12 @@ task("stageSdks") {
                 into(it.sourcesDir)
                 exclude { details ->
                     // Exclude `SmokeTests` directory
-                    details.file.name.endsWith("SmokeTests")
+                    val fileIsSmokeTests = details.file.name.endsWith("SmokeTests")
+
+                    // Exclude test files for internal clients
+                    val fileIsInternalClientGeneratedTests = it.internalClient && details.relativePath.startsWith("Tests")
+
+                    fileIsSmokeTests || fileIsInternalClientGeneratedTests
```

Comment:
Here: When copying the internal clients into the SDK, copy the whole project but leave out the `Tests/` folder.

This ensures the internal clients' dependencies are added to the project.